lib/copy/copy.js

```diff
@@ -113,23 +113,27 @@ async function onDir (srcStat, destStat, src, dest, opts) {
     await fs.mkdir(dest)
   }
 
-  const items = await fs.readdir(src)
+  const promises = []
 
   // loop through the files in the current directory to copy everything
-  await Promise.all(items.map(async item => {
-    const srcItem = path.join(src, item)
-    const destItem = path.join(dest, item)
+  for await (const item of await fs.opendir(src)) {
+    const srcItem = path.join(src, item.name)
+    const destItem = path.join(dest, item.name)
 
     // skip the item if it is matches by the filter function
     const include = await runFilter(srcItem, destItem, opts)
```

Comment:
Worth noting that if you passed an async filter function, this will now filter in series. This isn't super common, but might matter in some cases.

Reply:
Fixed in https://github.com/jprichardson/node-fs- extra/pull/1028/commits/f3e98087276c69269c89682eb487f87099074077.